sources/packages/api/src/services/application/application.service.ts

```diff
@@ -195,6 +197,82 @@ export class ApplicationService extends RecordDataModelService<Application> {
     );
   }
 
+  /**
+   * Updates Assessment status.
+   * @param applicationId application id to be updated.
+   * @param status status of the Assessment.
+   * An assessment need to happen, when the application_status is in assessment.
+   * @returns assessment status update result.
+   */
+  async updateAssessmentStatus(
+    applicationId: number,
+    status: AssessmentStatus,
+  ): Promise<UpdateResult> {
+    return this.repo.update(
+      { id: applicationId },
+      {
+        assessmentStatus: status,
+      },
+    );
+  }
+
+  /**
+   * Updates Confirmation of Enrollment(COE) status.
+   * @param applicationId application id to be updated.
+   * @param status status of the Confirmation of Enrollment.
+   * Confirmation of Enrollment need to happen, when the application_status is in enrollment.
+   * @returns COE status update result.
+   */
+  async updateCOEStatus(
+    applicationId: number,
+    status: COEStatus,
+  ): Promise<UpdateResult> {
+    return this.repo.update(
+      { id: applicationId },
+      {
+        coeStatus: status,
+      },
+    );
+  }
+
+  /**
+   * Updates overall Application status.
+   * @param applicationId application id to be updated.
+   * @param status status of the Application.
+   * @returns COE status update result.
+   */
+  async updateApplicationStatus(
+    applicationId: number,
+    status: ApplicationStatus,
+  ): Promise<UpdateResult> {
+    return this.repo.update(
+      { id: applicationId },
+      {
+        applicationStatus: status,
+      },
+    );
+  }
+
+  /**
+   * Updates overall Application status.
+   * @param applicationId application id to be updated.
+   * @param status status of the Application.
+   * @returns COE status update result.
+   */
+  async studentConfirmAssessment(
+    applicationId: number,
+    student: Student,
+  ): Promise<UpdateResult> {
+    return this.repo.update(
+      { id: applicationId, student: student },
```

Comment:
We don't need to use the entire Student object to select the application.
I would recommend receiving in the parameter only what is need, in this case, the student id.
The "where" would be like below.
```
return this.repo.update(
      { id: applicationId, student: { id: studentId }}
```